src/app/pages/BlocksPage/index.tsx

```diff
@@ -19,7 +19,30 @@ export const BlocksPage: FC = () => {
     {
       limit: PAGE_SIZE,
     },
-    { query: { refetchInterval: REFETCH_INTERVAL } },
+    {
+      query: {
+        refetchInterval: REFETCH_INTERVAL,
+        structuralSharing: (previousState, nextState) => {
+          if (!previousState) {
+            return nextState
+          }
+          return {
+            ...nextState,
+            data: {
+              blocks: nextState.data?.blocks?.map(block => {
+                const markAsNew = !previousState.data?.blocks?.some(
+                  prevBlock => prevBlock.round === block.round,
+                )
+                return {
+                  ...block,
+                  markAsNew,
```

Comment:
Unfortunately return type isn't updated :/

Reply:
OK I did that, too. (At the query definition itself, because if I do it on this function, the query type is not inferred correctly.)